draft-ietf-masque-connect-ip.md

```diff
@@ -443,6 +443,13 @@ IP proxies MAY perform access control using the scoping information provided
 by the client: if the client is not authorized to access any of the destinations
 included in the scope, then the IP proxy can immediately fail the request.
 
+Note that IP protocol numbers represent both upper layers (as defined in
+{{Section 2 of !IPv6=RFC8200}}, examples include TCP and UDP) and IPv6
+extension headers (as defined in {{Section 4 of IPv6}}, examples include
+Fragment and Routing headers). Implementations that support scoping by IP
+protocol number MUST either drop packets with IPv6 extension headers, or walk
+the chain of extensions to find the upper layer.
```

Comment:
Hm, I was expecting that our suggestion would be to reject scoping to a particular extension only.
```suggestion
Fragment and Routing headers). Implementations that support scoping by IP
protocol number MAY reject requests to scope to protocol numbers that are
used for extension headers. Upon receiving packets, implementations MUST
either drop packets that include IPv6 extension headers that don't match
the scoped protocol, or walk the chain of extensions to find the matching
protocol.
```

Reply:
I think we should remove the "drop packets" with extension headers notion. The only thing required by an intermediary is to be able to walk the chain of extension headers to the actual upper layer protocol. A connect-ip node MAY process hop-by-hop options. So this should be straight forward and remove the risk that the MASQUE endpoint contributes to blackholing IPv6 traffic. I would think we should try to follow https://www.rfc-editor.org/rfc/rfc8504.html#section-5.2 and consider the MASQUE endpoint as a IPv6 forwarding node. I do understand that there exist nodes on the internet that will drop packets with extension headers. 

Also, referencing Routing header is likely bad, consider this quote from RFC 8504: 
   An IPv6 node MUST be able to process these extension headers.  An
   exception is Routing Header type 0 (RH0), which was deprecated by
   [[RFC5095](https://www.rfc-editor.org/rfc/rfc5095)] due to security concerns and which MUST be treated as an
   unrecognized routing type.

Reply:
* I agree with Tommy that we should mention that disallowing scoping by extension header makes sense
* I agree with Magnus that Routing was a bad example - replacing with Options

```suggestion
Fragment and Options headers). IP proxies MAY reject requests to scope
to protocol numbers that are used for extension headers. Upon receiving
packets, implementations that support scoping by IP protocol number MUST
walk the chain of extensions to find the matching IP protocol number.
```